boundedint/int256_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package boundedint
+
+import (
+	"github.com/ethersphere/swarm/state"
+	"io/ioutil"
+	"math/big"
+	"testing"
+)
+
+var int256TestCases = []BoundedIntTestCase{
+	{
+		name:         "base 0",
+		baseInteger:  big.NewInt(0),
+		expectsError: false,
+	},
+	// negative numbers
+	{
+		name:         "base -1",
+		baseInteger:  big.NewInt(-1),
+		expectsError: false,
+	},
+	{
+		name:         "base -1 * 2^8",
+		baseInteger:  new(big.Int).Mul(big.NewInt(-1), new(big.Int).Exp(big.NewInt(2), big.NewInt(8), nil)),
```

Comment:
i would prefer to see hardcoded numbers here in sake of readability, and also to have in a comment in this test the exact values over which the int should over/underflow

Reply:
i cannot add the hard-coded value as an `int`, AFAIK it can only be added as a string (besides the current form).

given that we have the `name` field (which already references the value) is it still desirable to have a string name that includes all the digits?

in some cases this would be way too many digits to be humanly readable.

Reply:
i have added (hopefully) explanatory comments right above the test cases vars.

note that i kept the power notation of the interval bounds for the sake of readability, e.g.:

> 2^256

vs
> 115,792,089,237,316,195,423,570,985,008,687,907,853,269,984,665,640,564,039,457,584,007,913,129,639,936

Reply:
hm

Reply:
the name says `-1 * 2^8` which is `-1 * 256` so i'm not sure why you need to represent a full 2^256 here

Reply:
you're right, i don't need to call `Exp` for all the test cases. 

i've changed the test cases in which the literal number can be written